### ISSUE TITLE
Does auto-derived Getable only work on records and not Rust-originating structs?

### ISSUE THREAD

**Issue by t184256** [gluon-lang/gluon#591]
Sorry if this is trivial, but I can't get a derived Getable to return from gluon to Rust. First experience with Rust and gluon, yadda yadda.

To get something that is sure to work and to be on the same page with @Marwes and @Laegluin, I have adapted the marshalling example to this: https://gist.github.com/t184256/3d969d4bf862d124ed1948dea9042466

I added
```
fn make_struct(s: String) -> Struct {
    Struct {string: s, int: 5, tuple: (0.1, 0.2)}
}
```

and tried to return the result of `make_struct "Hello"` to Rust. It fails with

```
thread 'main' panicked at 'Cannot find the field 'string'. Do the type definitions match?'
```
at `<gluon_test::Struct as gluon_vm::api::Getable<'__vm>>::from_value`

and I have no idea how to make the struct return to Rust land.

Meanwhile, returning a record with the required fields present is totally fine.

So, does auto-derived Getable::from_value only work on records and not proper Rust-originating instances of the structs in question? How can I return a struct that I already shoved into Gluon succesfully, unmodified? Could you guys write trivial examples as well and not dive into overcomplicated ones right away?

**Comment by Marwes** [comment]
Thanks for reporting, seems like you ran into a bug with the Pushable derive, fix in #593

**Comment by t184256** [comment]
Great, specifying
```
gluon = { git = "https://github.com/Marwes/gluon", branch = "issue_591" }
gluon_vm = { git = "https://github.com/Marwes/gluon", branch = "issue_591" }
gluon_codegen = { git = "https://github.com/Marwes/gluon", branch = "issue_591" }
```
really fixed the problem I was having. Thanks!
